graphstorm-processing/tests/test_dist_numerical_transformation.py

```diff
@@ -0,0 +1,299 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import os
+
+import pytest
+import pandas as pd
+from numpy.testing import assert_array_equal, assert_array_almost_equal
+from pyspark.sql import SparkSession, DataFrame, functions as F
+from pyspark.sql.types import ArrayType, FloatType, StructField, StructType, StringType
+
+from graphstorm_processing.data_transformations.dist_transformations import (
+    DistNumericalTransformation,
+    DistMultiNumericalTransformation,
+)
+
+_ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+@pytest.fixture(scope="function", name="multi_num_df_with_missing")
+def multi_num_df_with_missing_fixture(spark: SparkSession):
+    """Create a multi-numerical text DataFrame with missing data"""
+    data = [("1|10",), ("5|2",), ("3|",), ("1|3",), ("5|2",)]
+
+    schema = StructType([StructField("ratings", StringType(), True)])
+    df = spark.createDataFrame(data, schema=schema)
+
+    yield df
+
+
+@pytest.fixture(scope="function", name="multi_num_df_without_missing")
+def multi_num_df_without_missing_fixture(spark: SparkSession):
+    """Create a multi-numerical text DataFrame without missing data"""
+    data = [("1|10",), ("5|2",), ("3|2",), ("1|3",), ("5|2",)]
+
+    schema = StructType([StructField("ratings", StringType(), True)])
+    df = spark.createDataFrame(data, schema=schema)
+
+    yield df
+
+
+def test_numerical_transformation_with_mode_imputer(input_df: DataFrame):
```

Comment:
Where is input_df?

Reply:
It's defined as an auto-use fixture in `conftest.py` to make it available to all test modules because we use it pretty much everywhere.